### ISSUE TITLE
Replace ButtonSelect widgets with radio buttons or select

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#9838]
### Issue Summary

Our filters use a `ButtonSelect` widget, which is a custom implementation of a form field that looks like a series of buttons but behaves like a radio button.

This was discussed during the last accessibility team meeting (Jan 2922) and also in [Stimulus project](https://github.com/orgs/wagtail/projects/14/views/1) discussions.

This JavaScript implementation creates the following problems;

- The "on" state isn’t communicated to screen reader users
- The "on" state is not visually indicated to WHCM users
- This widget feels very superfluous, clocking in at 100+ lines of code to make a worse version of radio buttons

### Steps to Reproduce

Go to the Reports views where the component is used in the filters.

<img width="1873" alt="Screenshot 2023-02-07 at 7 55 38 am" src="https://user-images.githubusercontent.com/1396140/217097017-14e4cc52-bb93-4f4c-94ac-caba66f2f8c8.png">

<img width="1892" alt="Screenshot 2023-02-07 at 8 02 19 am" src="https://user-images.githubusercontent.com/1396140/217097762-f5a9117e-287f-43b6-9c8f-e5c008ae9ca1.png">

### Recommended solution

Replace this with vanilla `input type="radio"` **or** `<select>` elements to fix existing accessibility issues with this widget.

* See the implementation of `ButtonSelect` - [wagtail/admin/widgets/button_select.py](https://github.com/wagtail/wagtail/blob/main/wagtail/admin/widgets/button_select.py)
* The styles can probably be deleted: https://github.com/wagtail/wagtail/blob/main/client/scss/components/_button-select.scss
* The HTML templates can be deleted wagtail/admin/templates/wagtailadmin/widgets/button_select.html & wagtail/admin/templates/wagtailadmin/widgets/button_select_option.html
* The JS can be deleted: client/src/includes/initButtonSelects.ts & client/src/includes/initButtonSelects.test.js
* The `ButtonSelect` widget can probably be deleted, but other usage needs to be considered, possibly just replacing with the suitable `forms.RadioSelect` or `forms.Select`.
* The `BooleanButtonSelect` can probably be renamed to `BooleanFilterSelect` - this widget override is still helpful as it causes 'nice' default values to show in the select options.
* Add note to the release notes (can be done on merge) that this widget will no longer be available. It is not documented but still good to call out.

#### Example screenshots

Recommended: radio - it looks better

**using `class ButtonSelect(forms.RadioSelect): ... pass`**

<img width="1891" alt="Screenshot 2023-02-07 at 8 04 05 am" src="https://user-images.githubusercontent.com/1396140/217098148-fbddcea8-57cc-4790-bba9-c773f5434eb3.png">

**using `class ButtonSelect(forms.Select): ... pass**

<img width="1874" alt="Screenshot 2023-02-07 at 8 03 47 am" src="https://user-images.githubusercontent.com/1396140/217098172-0c97e44c-a5c1-4c43-ad07-45a22502e5fb.png">

**Comment by Nazi-pikachu** [comment]
Hi @thibaudcolas I am new to open source and excited to work on this issue. Can i go for it?

**Comment by salty-ivy** [comment]
> Hi @thibaudcolas I am new to open source and excited to work on this issue. Can i go for it?

@Nazi-pikachu if you're a beginner I would suggest you to 
Start with issues labeled with good first issue, in this I would like to give this a try However if you feel that you can take this one up, then I can hold and you are much appreciated to have a try, thank you

**Comment by allcaps** [comment]
There are probably multiple implementations of list filters in Wagtail (ModelAdmin, Snippets, Pages(?)). Just wanted to mention that the Wagtail ModelAdmin uses [Django ModelAdmin List Filters](https://docs.djangoproject.com/en/4.1/ref/contrib/admin/filters/).

Django ModelAdmin List Filters are cool:
- They are well developed, and used in most Django projects.
- They ship with Django
- They are self contained (multiple filters can be used together, they don't need to be aware of each other). 
- They have good documentation, also on how to extend them.
- There are third party packages that allow adding more advanced filters. https://pypi.org/project/django-admin-list-filters/

I'm not sure about the accessibility of Django ModelAdmin List Filters, but if they are accessible, we might consider them???

An alternative option can be  [Django Filter](https://django-filter.readthedocs.io/) package.

**Comment by gasman** [comment]
Reports and generic index views (including snippets) are using django-filter, so with the ongoing work to position snippets as the successor to ModelAdmin, that will most likely emerge as the standard approach across Wagtail.

**Comment by lb-** [comment]
Thanks @allcaps  / @gasman  - we may end up splitting the 'source' behaviour change out to a separate issue, hoping to keep this issue focused on the simplest change possible; remove the JS implementation and replace with a simple HTML radio or select box.

**Comment by lb-** [comment]
Opening this up as a good first issue, this will require some Django/Python knowledge but not too much, mostly this requires deleting a bunch of files and renaming some.

cc @Nazi-pikachu  / @salty-ivy

**Comment by Pritimayasahoo** [comment]
Please assign me this issue I want to solve this issue

**Comment by lb-** [comment]
@Pritimayasahoo feel free to give this a start. We don't assign issues outside of the core team.

**Comment by Pritimayasahoo** [comment]
Can I solve this issue for your organization

**Comment by Astraxx04** [comment]
Heyy can I work on this issue. I have some knowledge on this and would like to work.

**Comment by Pritimayasahoo** [comment]
Can you tell me on which html file that html code exist, which you mention above in the screen short. Please help me I am a beginner, I want to solve this issue.

**Comment by salty-ivy** [comment]
> Can you tell me on which html file that html code exist, which you mention above in the screen short. Please help me I am a beginner, I want to solve this issue. 

@Pritimayasahoo you can join slack and ping me there, would help you through out your PR, also there would be other community members/ contributors you can get help from too.
In general you can ask these doubts there and on you PR and leave this space for ideas/discussions based on solutions or the issue itself

**Comment by lb-** [comment]
@Pritimayasahoo  - have another read of the description and follow every link, you will note there is a link to the `button_select.py` file - that Django Widget will declare what templates are used.

Specifically this line
https://github.com/wagtail/wagtail/blob/main/wagtail/admin/widgets/button_select.py#L11

Here are some docs on Django widgets also that may help https://docs.djangoproject.com/en/4.1/ref/forms/widgets/

**Comment by suyash5053** [comment]
./assign

**Comment by AceHunterr** [comment]
I'm trying to solve this issue and have created the radio button... but the problem I'm facing is where are those button clicks redirecting to? I'm trying to convert the radio button obtained into clickable links but don't know where to refer them? 

A little help would be appreciated

**Comment by AbhinavOhri** [comment]
I have opened a draft PR https://github.com/wagtail/wagtail/pull/10121 . Please review my work.

**Comment by AceHunterr** [comment]
I've tried solving this issue in PR #10123.If any issues are there please lemme know I'll correct them.

**Comment by lb-** [comment]
Removing good first issue - two PRs already up
